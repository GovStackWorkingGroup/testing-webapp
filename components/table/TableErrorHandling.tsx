import React, { useCallback } from 'react';
import { useIntl } from 'react-intl';

const TableErrorHandling = () => {
  const { formatMessage } = useIntl();
  const format = useCallback(
    (id: string) => formatMessage({ id }),
    [formatMessage]
  );

  return (
    <div className='table-error-handling' data-testid='table-error'>
      <p>{format('app.error_fetching_data.message')}</p>
    </div>
  );
};

export default TableErrorHandling;
