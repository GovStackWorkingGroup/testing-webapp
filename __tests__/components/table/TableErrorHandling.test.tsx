import { render } from '../../test-utils/test-utils';
import TableErrorHandling from '../../../components/table/TableErrorHandling';

describe('Unit tests for TableErrorHandling component:', () => {
  const MESSAGE_TEST_ID = 'table-error';

  it('should render the SubTable TableErrorHandling and match snapshot', () => {
    const { container } = render(<TableErrorHandling />);

    expect(container).toMatchSnapshot();
  });

  it('when have corresponding values', () => {
    const { getByTestId } = render(<TableErrorHandling />);

    expect(getByTestId(MESSAGE_TEST_ID)).toHaveTextContent(
      'Error fetching data. Please try again later.'
    );
  });
});
