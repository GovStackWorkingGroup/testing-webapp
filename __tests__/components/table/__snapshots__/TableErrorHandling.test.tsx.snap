// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for TableErrorHandling component: should render the SubTable TableErrorHandling and match snapshot 1`] = `
<div>
  <div
    class="table-error-handling"
    data-testid="table-error"
  >
    <p>
      Error fetching data. Please try again later.
    </p>
  </div>
</div>
`;
